modules/llrt_buffer/src/buffer.rs

```diff
@@ -120,6 +120,25 @@ fn byte_length<'js>(ctx: Ctx<'js>, value: Value<'js>, encoding: Opt<String>) ->
     ))
 }
 
+fn is_buffer<'js>(ctx: Ctx<'js>, value: Value<'js>) -> Result<bool> {
+    let class: Function = ctx.globals().get(stringify!(Buffer))?;
+    if let Some(object) = value.as_object() {
+        return Ok(object.is_instance_of(class));
+    }
+
+    Ok(false)
+}
+
+fn is_encoding(value: Value) -> Result<bool> {
+    if let Some(js_string) = value.as_string() {
+        if let Ok(std_string) = js_string.to_string() {
```

Comment:
We should fail here if we can't convert the JsString to rust string
```rust
let string = js_string.to_string()?;
```

Reply:
Good catch, addressed in next commit